cpp/modmesh/view/RCameraController.hpp

```diff
@@ -135,30 +135,59 @@ class CameraController
 
     virtual void updateCameraPosition(const CameraInputState & state, float dt) = 0;
 
-    virtual Qt3DRender::QCamera * getCamera() = 0;
+    virtual Qt3DRender::QCamera * camera() const = 0;
+    virtual void setCamera(Qt3DRender::QCamera * camera) = 0;
 
-    virtual float getLinearSpeed() = 0;
+    virtual float linearSpeed() const = 0;
+    virtual void setLinearSpeed(float value) = 0;
 
-    virtual float getLookSpeed() = 0;
+    virtual float lookSpeed() const = 0;
+    virtual void setLookSpeed(float value) = 0;
 
     virtual CameraControllerType getType() = 0;
 
-    QVector3D position() { return getCamera()->position(); }
+    QVector3D position() const { return camera()->position(); }
+    void setPosition(const QVector3D & value) const { camera()->setPosition(value); }
 
-    QVector3D viewVector() { return getCamera()->viewVector(); }
+    QVector3D viewVector() const { return camera()->viewVector(); }
 
-    QVector3D viewCenter() { return getCamera()->viewCenter(); }
+    QVector3D viewCenter() const { return camera()->viewCenter(); }
+    void setViewCenter(const QVector3D & value) const { camera()->setViewCenter(value); }
 
-    QVector3D upVector() { return getCamera()->upVector(); }
+    QVector3D upVector() const { return camera()->upVector(); }
+    void setUpVector(const QVector3D & value) const { camera()->setUpVector(value); }
 
-protected:
-    RCameraInputListener * m_listener = nullptr;
+    void reset();
+
+    QVector3D defaultPosition() const { return m_default_position; }
+    void setDefaultPosition(QVector3D value) { m_default_position = value; }
+
+    QVector3D defaultViewCenter() const { return m_default_view_center; }
+    void setDefaultViewCenter(QVector3D value) { m_default_view_center = value; }
+
+    QVector3D defaultUpVector() const { return m_default_up_vector; }
+    void setDefaultUpVector(QVector3D value) { m_default_up_vector = value; }
+
+    float defaultLinearSpeed() const { return m_default_linear_speed; }
+    void setDefaultLinearSpeed(float value) { m_default_linear_speed = value; }
+
+    float defaultLookSpeed() const { return m_default_look_speed; }
+    void setDefaultLookSpeed(float value) { m_default_look_speed = value; }
 
-private:
     Qt3DExtras::QAbstractCameraController * asQtCameraController()
     {
         return dynamic_cast<Qt3DExtras::QAbstractCameraController *>(this);
     }
+
+protected:
+    RCameraInputListener * m_listener = nullptr;
+
+private:
+    QVector3D m_default_position = QVector3D(0.0f, 0.0f, 10.0f);
```

Comment:
why (0, 0, 10)?

Reply:
Because mesh are placed at `(0, 0, 0)`, and we don't want the camera to be inside the mesh

Reply:
I can guess that, but my point is, better to have some comments for it.

Reply:
>> Because meshes are placed at `(0, 0, 0)`, and we don't want the camera to be inside the mesh

To put the camera outside the mesh, it is not good to hard-code the default value of the member `m_default_position`, because the mesh may be placed anywhere.

We do not have a reasonably large number of mesh examples for the default location to show disadvantage.

> I can guess that, but my point is, better to have some comments for it.

I concur, but it's OK to leave it in the future.  As we make the code more useful, some meshes will cover the position sooner or later.  We will immediately know and fix it.